src/Model/Model.php

```diff
@@ -436,10 +436,11 @@ protected function prepare_fields() {
 		 * @param string   $visibility   The visibility setting for this piece of data
 		 * @param null|int $owner        The user ID for the owner of this piece of data
 		 * @param \WP_User $current_user The current user for the session
+		 * @param object   $data         The raw data passed into the model
 		 *
 		 * @return array
 		 */
-		$this->fields = apply_filters( 'graphql_return_modeled_data', $this->fields, $this->get_model_name(), $this->visibility, $this->owner, $this->current_user );
+		$this->fields = apply_filters( 'graphql_return_modeled_data', $this->fields, $this->get_model_name(), $this->visibility, $this->owner, $this->current_user, $this->data );
```

Comment:
I believe we still need a few more filters in this class, eh?

Like `$this->owner` I think needs to be filterable. For example, Orders in WooCommerce are not "owned" by the `post->post_author` so the `owner` needs to be filters for `Posts` of `$this->data->post_type === 'Order'`

Reply:
I don't think so. I think people will just want to use the filters in the `get_visibility` method: https://github.com/wp-graphql/wp-graphql/blob/dbdf14f5b6adb12d8343f9d3b69cb7a96c7a3b3c/src/Model/Model.php#L187-L242

Reply:
@CodeProKid I think it probably still should be filtered early when it's set. 

`$this->owner` gets passed to a lot of filters, and if it's not accurate (like the case of WooCommerce orders), how should these other filters use it?

If the `__construct` did something like: 

```
$this->owner = apply_filters( 'graphql_model_owner', $owner, ...$other_args );
```

Then something like WooGraphQL could filter the owner based on the `post_type === order`, and all other filters that get `$owner` as an argument would have it already properly filtered for use within their filtering. 

There could easily be a case where filtering fields might need to take into granular details about the `$owner` to decide how to resolve a field, for example.

Reply:
@jasonbahl I'm a little hesitant to add a filter in the `__construct` since a lot of the data hasn't been setup by then, so I'm not sure how useful it would be. I ended up adding a filter to the `owner_matches_current_user` method in this commit: https://github.com/wp-graphql/wp-graphql/pull/1281/commits/ea0376a53ee9c4ab61bff9124c77e80abac03db7 LMK if you think that still wouldn't suffice and we can revisit.